Livepeer.Cloud SPE: Add support for NEXT_PUBLIC_AI_METRICS_SERVER_URL to enable separate AI metrics endpoint

mikezupper opened the pull request:
This pull request introduces support for a new environment variable: `NEXT_PUBLIC_AI_METRICS_SERVER_URL`. This enables Livepeer Explorer deployments to configure a separate metrics server for AI workloads, independent of the existing `NEXT_PUBLIC_METRICS_SERVER_URL` used for traditional transcoding metrics.

- **Environment Configuration**
  - Added `NEXT_PUBLIC_AI_METRICS_SERVER_URL` to:
    - `.env.example`
    - `README.md`
    - `Dockerfile` for build-time variable injection

- **API Updates**
  - `/api/score/[address].ts`: Uses AI metrics server for top score queries
  - `/api/pipelines/index.ts`: Fetches AI pipelines from the AI metrics server
  - `/api/regions/index.ts`: Supports querying both metrics server URLs

This change is **backward-compatible**. If `NEXT_PUBLIC_AI_METRICS_SERVER_URL` is not defined, only the original `NEXT_PUBLIC_METRICS_SERVER_URL` is used. Existing deployments focused solely on transcoding metrics are unaffected.

### Deployment Note

Be sure to update your `.env` file to include the new variable if deploying with AI metrics:

```env
NEXT_PUBLIC_AI_METRICS_SERVER_URL=https://your-ai-metrics-server-url

0xspeedybird committed:
add flag to use Livepeer.Cloud SPE AI Metrics API

0xspeedybird committed:
Update to merge regions from both the transcoding and ai metrics API endpoints as well as use the approriate server when retrieving scores.

rickstaa committed:
refactor: small code cleanup

rickstaa reviewed (approved):
@0xspeedybird I made some small changes but looks good to me.

mikezupper commented:
LGTM!

rickstaa committed:
fix: improve orch score error handling

The old logic caused the explorer to crash if the stats endpoint
returned an error. The new logic handles these errors gracefully.

rickstaa committed:
Merge branch 'main' into tasks/livepeer.cloud/use-livepeer-cloud-ai-metrics
